From b1a9c69792caaac0813c9dc7806fb8654212e56c Mon Sep 17 00:00:00 2001
From: Michael Shych <michaelsh@nvidia.com>
Date: Sat, 30 Apr 2022 14:58:09 +0300
Subject: [PATCH] Documentation/ABI: Add new attributes for mlxreg-io sysfs
 interfaces

Add documentation for the new attributes:
- "phy_reset" - Reset PHY.
- "mac_reset" - Reset MAC.
- "qsfp_pwr_good" - The power status of QSFP ports.

Signed-off-by: Michael Shych <michaelsh@nvidia.com>
Reviewed-by: Vadim Pasternak <vadimp@nvidia.com>
Link: https://lore.kernel.org/r/20220430115809.54565-4-michaelsh@nvidia.com
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---

diff --git a/Documentation/ABI/stable/sysfs-driver-mlxreg-io b/Documentation/ABI/stable/sysfs-driver-mlxreg-io
index 12c3f895cd2f..b312242d4f40 100644
--- a/Documentation/ABI/stable/sysfs-driver-mlxreg-io
+++ b/Documentation/ABI/stable/sysfs-driver-mlxreg-io
@@ -457,7 +457,7 @@
 
 		The files are read only.
 
-What:	/sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/asic1_reset
+What:		/sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/asic1_reset
 		/sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/asic2_reset
 Date:		October 2021
 KernelVersion:	5.16
@@ -470,21 +470,57 @@
 
 		The files are write only.
 
-What:                    /sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/thshtdwn_unlock
-Date:                     October 2021
+What:		/sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/thshtdwn_unlock
+Date:		October 2021
 KernelVersion:  5.16
-Contact:    Vadim Pasternak vadimp@nvidia.com
+Contact:	Vadim Pasternak vadimp@nvidia.com
 Description:    This file allows to unlock ASIC after shutdown event.
-        When system thermal shutdown is enforced by ASIC, ASIC
-        is getting locked and after system boot it will not be
-        available. Software can decide to unlock it by setting
-        this attribute to 1 and then perform system power cycle
-        by setting “pwr_cycle” attribute to 1 (power cycle of
-        main power domain).
-        Prior setting “thshtdwn_unlock” 1 it is recommended to
-        validate that system reboot cause is “reset_asic_thermal”
-        In case “thshtdwn_unlock” is not set 1, the only way to
-        release ASIC from locking - is full system power cycle
-        through the external power distribution unit.
+		When system thermal shutdown is enforced by ASIC, ASIC
+		is getting locked and after system boot it will not be
+		available. Software can decide to unlock it by setting
+		this attribute to 1 and then perform system power cycle
+		by setting "pwr_cycle" attribute to 1 (power cycle of
+		main power domain).
+		Prior setting "thshtdwn_unlock" 1 it is recommended to
+		validate that system reboot cause is "reset_asic_thermal"
+		In case "thshtdwn_unlock" is not set 1, the only way to
+		release ASIC from locking - is full system power cycle
+		through the external power distribution unit.
+
+		The files are write only.
 
+What:           /sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/phy_reset
+Date:		May 2022
+KernelVersion:	5.19
+Contact:        Vadim Pasternak <vadimpmellanox.com>
+Description:    This file allows to reset PHY 88E1548 when attribute is set 0
+                due to some abnormal PHY behavior.
+                Expected behavior:
+                When phy_reset is written 1, all PHY 88E1548 are released
+                from the reset state, when 0 - are hold in reset state.
+
+                The files are read/write.
+
+What:           /sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/mac_reset
+Date:		May 2022
+KernelVersion:	5.19
+Contact:        Vadim Pasternak <vadimpmellanox.com>
+Description:    This file allows to reset ASIC MT52132 when attribute is set 0
+                due to some abnormal ASIC behavior.
+                Expected behavior:
+                When mac_reset is written 1, the ASIC MT52132 is released
+                from the reset state, when 0 - is hold in reset state.
+
+                The files are read/write.
+
+What:           /sys/devices/platform/mlxplat/mlxreg-io/hwmon/hwmon*/qsfp_pwr_good
+Date:		May 2022
+KernelVersion:	5.19
+Contact:        Vadim Pasternak <vadimpmellanox.com>
+Description:    This file shows QSFP ports power status. The value is set to 0
+                when one of any QSFP ports is plugged. The value is set to 1 when
+                there are no any QSFP ports are plugged.
+                The possible values are:
+                0 - Power good, 1 - Not power good.
 
+                The files are read only.
